userspace/libsinsp/filterchecks.cpp

```diff
@@ -6707,7 +6708,25 @@ uint8_t* sinsp_filter_check_container::extract(sinsp_evt *evt, OUT uint32_t* len
 			m_tstr = "NONE";
 			RETURN_EXTRACT_STRING(m_tstr);
 		}
-
+	case TYPE_CONTAINER_IP_NAME:
+		if(tinfo->m_container_id.empty())
+		{
+			return NULL;
+		}
+		else
+		{
+			const sinsp_container_info::ptr_t container_info =
+				m_inspector->m_container_manager.get_container(tinfo->m_container_id);
+			if(!container_info)
+			{
+				return NULL;
+			}
+			m_u32val = htonl(container_info->m_container_ip);
+			char addrbuff[100];
+			inet_ntop(AF_INET, &m_u32val, addrbuff, sizeof(addrbuff));
```

Comment:
Hmm... do we always assume IPv4 here?

Reply:
We don't really track container ipv6 addresses I think, `m_container_ip` is always ipv4

Reply:
Let's check if something could be done about ipv6. If only ipv4 is possible in the near term, the `container.ip` definition should reflect this fact ... Looking into this this coming week and will then rename `TYPE_CONTAINER_IP_NAME` to  `TYPE_CONTAINER_IP_ADDR`.

Reply:
Hi @hbrueckner and @gnosek briefly synced up with Michael Crosby from the containerd project and we both converged on that logging the raw unparsed JSON string of the CRI PodSandboxStatusResponse info.cniResult.Interfaces would provide the most complete picture / most useful information for incident response.

Only did limited testing so far (on localhost with some pods launched other crictl). What would you think of this approach? However, would also still keep container.ip.

Updated the initial comment of this PR.

Reply:
And also only implemented and tested it for `containerd` ... hence more work would be needed, but wanted to first check in if this is in the right direction.

Reply:
@incertum Did you have the chance to also test with other container engine?

Reply:
> We don't really track container ipv6 addresses I think, `m_container_ip` is always ipv4

Thanks for clarification and looking at the definitions it becomes clear. Thanks to @incertum to update the documentation in this regard. However, as you pointed out, there is some way to get this information from the CNI json result.